src/photon/photon_algorithm.c

```diff
@@ -143,9 +217,11 @@ bool can_run(scheduling_algorithm_state *algorithm_state, task_spec *task) {
 
 /* TODO(rkn): This method will need to be changed to call reconstruct. */
 int fetch_object_timeout_handler(event_loop *loop, timer_id id, void *context) {
-  fetch_object_request *fetch_req = (fetch_object_request *) context;
-  object_id object_ids[1] = {fetch_req->object_id};
-  plasma_fetch(fetch_req->state->plasma_conn, 1, object_ids);
+  local_scheduler_state *state = context;
+  object_entry *fetch_request, *tmp;
+  HASH_ITER(hh, state->algorithm_state->fetch_requests, fetch_request, tmp) {
+    plasma_fetch(state->plasma_conn, 1, &fetch_request->object_id);
```

Comment:
`plasma_fetch` actually takes an array of object IDs, so maybe we should batch these together.